pkg/build/controller/controller.go

```diff
@@ -125,29 +125,8 @@ func (bc *BuildController) nextBuildPhase(build *buildapi.Build) error {
 	// Set the build phase, which will be persisted if no error occurs.
 	build.Status.Phase = buildapi.BuildPhasePending
 
-	// Make a copy to avoid mutating the build from this point on.
```

Comment:
It is in the scope of this PR to make sure we don't change the build in the builders. (FYI @PI-Victor)

For sure we cannot persist the changes, the updater code explicitly ensures that the build.spec is not changed, and returns an error if it was (that's why we were making a copy here).

I think CreateBuildPod does not change the build, but certainly there is code in the builders (s2i and docker) that does (even thought it is not persistent, but IMHO still wrong). @PI-Victor should be fixing those in this PR.
(e.g. https://github.com/openshift/origin/pull/5071/files#diff-b9627ec6ab0689e50e5fa3e5d958c1d3L83)